compiler/injectdestructors.nim

```diff
@@ -1153,7 +1153,8 @@ proc moveOrCopy(dest, ri: PNode; c: var Con; s: var Scope, flags: set[MoveOrCopy
         let snk = c.genSink(s, dest, ri, flags)
         result = newTree(nkStmtList, snk, c.genWasMoved(ri))
       elif ri.sym.kind != skParam and ri.sym.owner == c.owner and
-          isLastRead(ri, c, s) and canBeMoved(c, dest.typ) and not isCursor(ri):
+          isLastRead(ri, c, s) and canBeMoved(c, dest.typ) and not isCursor(ri) and
+          {sfGlobal, sfPure} <= ri.sym.flags == false:
```

Comment:
```suggestion
          {sfGlobal, sfPure} * ri.sym.flags == {}:
```

Reply:
it's for "not a subset of ri.sym.flags", I know the code too long, I have to revisit my old patch that mis use `*`

Reply:
Well, you are free to use `ri.sym.flags < {sfGlobal, sfPure}` or `{sfGlobal, sfPure} * ri.sym.flags != {sfGlobal, sfPure}:`.

Reply:
the last one looks ugly

Reply:
```nim
type E = enum
  a = 1
  b = 2
  c = 3
  e = 4

echo {a,b,c} < {c, e} # false
```
likely we need sugar operator.

Reply:
Is this some kind of joke? You merge your own PRs that have non resolved points of discussion?

Reply:
the final choice is `not ({sfGlobal, sfPure} <= ri.sym.flags)`

Reply:
That is not what "github diff" showed me but ok.

Reply:
That diff is from another later PR  https://github.com/nim-lang/Nim/commit/f18e4c4050cb59cf828372f89c01d9e80f6516c5